src/plugins/analysis/device_tree/view/device_tree.html

```diff
@@ -1,22 +1,83 @@
+    <script src="{{ url_for('static', filename='highlight.js/languages/dts.min.js') }}"></script>
+    <script>
+        async function async_highlight(element) {
+            element.innerHTML = hljs.highlight(
+                element.innerHTML.replaceAll("&lt;", "<").replaceAll("&gt;", ">"),
+                {language: 'dts'}
+            ).value;
+        };
+        document.addEventListener('DOMContentLoaded', (event) => {
+            document.querySelectorAll('.dt-pre').forEach((element) => {
+                async_highlight(element);
+            });
+        });
+    </script>
+{% endblock %}
+
+
 {% block analysis_result_details %}
 
-{% for key, analysis_data in firmware.processed_analysis[selected_analysis].items() | sort %}
-    {% if key | is_not_mandatory_analysis_entry %}
-        <tr>
-            <td>{{ key }}</td>
-            <td>
-                <pre class="border rounded p-2 bg-light" style="white-space: pre-wrap;">
-                    {% if key == 'device_tree' %}
-                        <code>{{ analysis_data | hide_dts_binary_data | nice_generic | safe }}</code>
-                    {% else %}
-                        <code>{{ analysis_data | nice_generic | safe }}</code>
-                    {% endif %}
-                </pre>
-            </td>
-        </tr>
-    {% endif %}
+{# Don't break on legacy result #}
+{% if "device_tree" in firmware.processed_analysis[selected_analysis] %}
+    <tr class="table-warning">
+        <td> Deprecation Warning </td>
+        <td> Analysis version is deprecated, please update the analysis </td>
+    </tr>
+    <tr>
+        <td> Device Tree </td>
+        {{ device_tree_table_cell(firmware.processed_analysis[selected_analysis].device_tree, "1") }}
+    </tr>
+{% endif %}
+
+{% for dt_dict in firmware.processed_analysis[selected_analysis].get("device_trees", []) %}
+    <tr data-toggle="collapse" data-target="#collapse-{{ dt_dict.offset }}">
+        <td class="clickable" colspan=2>
+            <i class="fas fa-angle-down"></i>
+            Offset {{ dt_dict.offset | hex }} | {{ dt_dict.model or dt_dict.description or "" }}
+        </td>
+    </tr>
+    <tr>
+        <td colspan="2" class="qemu-td p-0">
+            <div class="collapse" id="collapse-{{ dt_dict.offset }}">
+                <table class="table table-bordered table-sm m-0" style="width: 100%">
+                    {% for key, value in dt_dict.header.items() -%}
+                        <tr>
+                            <td width="200px">{{ key | replace_underscore }}</td>
+                            <td width="125px">{{ value | nice_number }}</td>
+                            <td>{{ value | hex }}</td>
+                        </tr>
+                    {%- endfor %}
+                    <tr>
+                        {{ device_tree_table_cell(dt_dict.device_tree, "3") }}
+                    </tr>
+                </table>
+            </div>
+        </td>
+    </tr>
 {% endfor %}
 
 {% endblock %}
```

Comment:
That looks really cool! And it appears to work as expected. Thank you very much.

I understand that the summary table is part of the base plugin.. but could we somehow switch it off? I mean it is kind of obsolete because of the collapsible. Is that easily possible? What do you think? Maybe we can discuss about that next monday :-)

Reply:
For the record: we just talked about that one. The summary does only exist on Firmware-Level. This was basically my fault :-)